code/modules/integrated_electronics/subtypes/manipulation.dm

```diff
@@ -304,19 +304,28 @@
 	activators = list("pulse in" = IC_PINTYPE_PULSE_IN,"pulse out" = IC_PINTYPE_PULSE_OUT)
 	spawn_flags = IC_SPAWN_RESEARCH
 	power_draw_per_use = 50
-	var/max_w_class = WEIGHT_CLASS_NORMAL
 	var/max_items = 10
-
 /obj/item/integrated_circuit/manipulation/grabber/do_work()
+	var/max_w_class = assembly.w_class
 	var/atom/movable/acting_object = get_object()
 	var/turf/T = get_turf(acting_object)
 	var/obj/item/AM = get_pin_data_as_type(IC_INPUT, 1, /obj/item)
 	if(AM)
 		var/mode = get_pin_data(IC_INPUT, 2)
-
 		if(mode == 1)
 			if(check_target(AM))
-				if((contents.len < max_items) && (!max_w_class || AM.w_class <= max_w_class))
+				var/weightcheck = FALSE
+				if ((!istype(AM,/obj/item/device/electronic_assembly/)) && (!istype(AM,/obj/item/device/transfer_valve)))
+					if (AM.w_class <= max_w_class)
+						weightcheck = TRUE
+					else
+						weightcheck = FALSE
+				else
+					if (AM.w_class < max_w_class)
+						weightcheck = TRUE
+					else
+						weightcheck = FALSE
+				if((contents.len < max_items) && (weightcheck))
```

Comment:
just leave there 
if((contents.len < max_items) && (!max_w_class || AM.w_class < max_w_class))
Large objects still sufficient. But better to have little more restrictions, than hardcoded list of restricted items.

Reply:
what the FUCK are large items? can i get at least one example of a large object that ISN'T just a normal sized object

Reply:
also, that would mean grabbers in normal sized circuit would be able to only store small items, and grabbers in small sized circuit wouldn't be able tto store anything at all, which i'm not really for

Reply:
oh fuck, you are right.
BTW, now you can't even stick grabber into small circuit assembly.
Which is fully logical.